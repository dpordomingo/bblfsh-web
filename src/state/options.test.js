import {
  reducer,
  initialState,
  LOCATIONS_TOGGLE,
  CUSTOM_SERVER_URL_SET,
  CUSTOM_SERVER_TOGGLE,
} from './options';

describe('options/reducer', () => {
  it('LOCATIONS_TOGGLE', () => {
    expect(
      reducer(
        { ...initialState, showLocations: true },
        { type: LOCATIONS_TOGGLE }
      )
    ).toMatchSnapshot();
    expect(reducer(initialState, { type: LOCATIONS_TOGGLE })).toMatchSnapshot();
  });

  it('CUSTOM_SERVER_URL_SET', () => {
    expect(
      reducer(initialState, {
        type: CUSTOM_SERVER_URL_SET,
        url: 'newUrl',
      })
    ).toMatchSnapshot();
  });

  it('CUSTOM_SERVER_TOGGLE', () => {
    expect(
      reducer(initialState, { type: CUSTOM_SERVER_TOGGLE })
    ).toMatchSnapshot();
    expect(
      reducer(
        { ...initialState, customServer: true },
        { type: CUSTOM_SERVER_TOGGLE }
      )
    ).toMatchSnapshot();
  });
});
