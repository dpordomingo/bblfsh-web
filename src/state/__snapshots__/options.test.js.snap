// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options/reducer CUSTOM_SERVER_TOGGLE 1`] = `
Object {
  "customServer": true,
  "customServerUrl": "0.0.0.0:9432",
  "showLocations": false,
}
`;

exports[`options/reducer CUSTOM_SERVER_TOGGLE 2`] = `
Object {
  "customServer": false,
  "customServerUrl": "",
  "showLocations": false,
}
`;

exports[`options/reducer CUSTOM_SERVER_URL_SET 1`] = `
Object {
  "customServer": false,
  "customServerUrl": "newUrl",
  "showLocations": false,
}
`;

exports[`options/reducer LOCATIONS_TOGGLE 1`] = `
Object {
  "customServer": false,
  "customServerUrl": "",
  "showLocations": false,
}
`;

exports[`options/reducer LOCATIONS_TOGGLE 2`] = `
Object {
  "customServer": false,
  "customServerUrl": "",
  "showLocations": true,
}
`;
